compaction/compaction_manager.cc

```diff
@@ -617,8 +635,11 @@ void compaction_manager::do_stop() noexcept {
 }
 
 inline bool compaction_manager::can_proceed(const lw_shared_ptr<task>& task) {
-    return (_state == state::enabled) && !task->stopping &&
-        (task->type != sstables::compaction_type::Compaction || !_compaction_state[task->compacting_cf].compaction_disabled());
+    if (_state != state::enabled || task->stopping) {
+        return false;
+    }
+    auto csptr = try_get_compaction_state(task->compacting_cf);
+    return csptr && (task->type != sstables::compaction_type::Compaction || !csptr->compaction_disabled());
```

Comment:
let's add a function in manager for table to register itself in its ctor, then we can drop get_compaction_state(), and only use try_get_compaction_state() which can be renamed to get_compaction_state().

Reply:
ok, I'll see if I can easily fit that in for the next version.
It sounds like a good idea.

Reply:
added in v2